File: ros2_foxy_tutorial_ws/src/my_cpp_pkg_/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"
 
class NumCountNode : public rclcpp::Node 
{
public:
    NumCountNode() : Node("number_counter"), counter_(0) 
    {
        subscriber_ = this->create_subscription<example_interfaces::msg::Int64>(
            "number", 10, 
            std::bind(&NumCountNode::callbackNumCount, this, std::placeholders::_1));
        
        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number_count", 10);
        
        RCLCPP_INFO(this->get_logger(), "Number counter has been started.");
    }

private:
    void callbackNumCount(const example_interfaces::msg::Int64::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(), "%ld", msg->data);
        
        this->counter_ += msg->data;
        auto msg_num = example_interfaces::msg::Int64();
        msg_num.data = this->counter_;
        publisher_->publish(msg_num);
    }

    std::int64_t counter_;
    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscriber_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;

};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumCountNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_foxy_tutorial_ws/src/my_py_pkg/my_py_pkg/number_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
 
from example_interfaces.msg import Int64

class NumCountNone(Node): 
    def __init__(self):
        super().__init__("number_counter")
        self.counter_ = 0
        self.subscriber_ = self.create_subscription(
            Int64, "number", self.callback_number_counter, 10)
        self.publisher_ = self.create_publisher(Int64, "number_count", 10)
        self.get_logger().info("Number counter has been started.")
 
    def callback_number_counter(self, msg):
        self.counter_ += msg.data
        self.get_logger().info(str(msg.data))

        msg = Int64()
        msg.data = self.counter_
        self.publisher_.publish(msg)
    
def main(args=None):
    rclpy.init(args=args)
    node = NumCountNone() 
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()